### PR TITLE
Addressed dp overlap with mem, compute slowdown

### PR DESCRIPTION
**Pull request by michael-isaev**

### CODE REVIEW

File: calculon/megatron/megatron.py [hunk 1]
```diff
@@ -114,6 +114,7 @@ def __init__(self, cfg):
       self._pipeline_par_rs_ag = \
         self.tensor_par_comm_type in ['p2p_rs_ag', 'rs_ag']
       self.data_par_overlap = cfg['data_par_overlap']
+      self.data_par_comm_resource = cfg['data_par_comm_resource']
```
Comment (nicmcd) [inline calculon/megatron/megatron.py:117]: This needs a range check
should be >= 0.0 and < 1.0

File: calculon/megatron/optimal_execution.py [hunk 2]
```diff
@@ -90,6 +90,7 @@ def search(debug, num_procs, max_batch_size, app, syst, tp, pp):
                               'tensor_par_comm_type': tensor_par_comm_type,
                               'seq_par_ag_redo': seq_par_ag_redo,
                               'data_par_overlap': data_par_overlap,
+                              'data_par_comm_resource': 0.02,
```
Comment (nicmcd) [inline calculon/megatron/optimal_execution.py:93]: This doesn't really work in this case. The optimal execution engine tries all networks for DP. If DP gets mapped to network 1 (i.e., NVLink) then the # of SMs would be larger (e.g., 8-16 ish). However, if it gets mapped to network 2 (i.e., InfiniBand) then the # of SMs would be small (e.g. 2).

**Review by nicmcd — CHANGES_REQUESTED** [review]

File: calculon/megatron/optimal_execution.py (hunk 2 above)
Comment (michael-isaev) [reply calculon/megatron/optimal_execution.py:93]: Oh, that's a great catch!
I had an idea that if you need 2 SMs for 50 GB/s IB less efficiency, then you should use multiply more SMs for NVLink 450 GB/s, less efficiency. You could probably have some scale stemming from some system parameter, e.g. how much bandwidth can single SM utilize, but that's not information that's easy to get - I didn't see many such benchmark examples.
I can transfer this parameter to system/network and call it something like `proc_utilization`

File: calculon/megatron/megatron.py (hunk 1 above)
Comment (michael-isaev) [reply calculon/megatron/megatron.py:117]: agreed, will do

File: calculon/megatron/optimal_execution.py (hunk 2 above)
Comment (nicmcd) [reply calculon/megatron/optimal_execution.py:93]: Wait on the changes, lets talk next week

Comment (nicmcd) [reply calculon/megatron/optimal_execution.py:93]: After further though, I think this should be a member of the system configuration listed under each network.

**Review by nicmcd — APPROVED** [review]

**Pull request merged**